### DISCUSSION TITLE
Droping large sysex messages

### DISCUSSION THREAD

**Discussion by wylie39** [djipco/webmidi#451] [Q&A]
I am working on a project that transfers project data from a MIDI device, I am hitting a roadblock as it seems `addListener("sysex")` will drop large messages. 
This is my test code:
```
WebMidi
  .enable({sysex: true})
  .then(() => {
    const myInput = WebMidi.getInputByName(deviceName);
    myInput.addListener("sysex", e => {
      console.log(e.rawData.length);
    })
  })
  .catch(err => alert(err));
```
It works fine for any small sysex messages, but if I try to receive a large one say 10,000 bytes it seems to get dropped, I can see that they are being received by MIDI monitor.
I have a hunch that it is only a problem on nodejs as when I use this test code in the browser it works as expected
```
navigator.requestMIDIAccess({sysex: true}).then(onMIDISuccess, onMIDIFailure);

function onMIDISuccess(midiAccess) {

    console.log("sysex enabled: ", midiAccess.sysexEnabled);

    for (var input of midiAccess.inputs.values()) {
        input.onmidimessage = message => console.log(message);
   }

}

function onMIDIFailure() {
    console.log('Could not access your MIDI devices.');
}
```

Any help would be appreciated 

Just in case it is helpful:
Library version: 3.1.11
MacOS: 14.5 (23F79)
Nodejs: v16.15.1 tested on v18.19.1 as well

**Comment by djipco** [comment]
WEBMIDI.js has been created for the browser first. Support for Node.js was added later by piggy-backing on the [`jzz`](https://github.com/jazz-soft/JZZ/) module. As of today, we have [3 pending issues](https://github.com/djipco/webmidi/issues) that I suspect are caused by this module (all are Node-related). If your problem only occurs on Node, it seems very likely that it is caused by this external module. Unfortunately, the team behind `jzz` has not been very reactive. If this persists, we may have to move Node.js support to something else... I wish I could be more helpful.

**Reply by wylie39** [reply]
That's sort of what I expected, I will poke around the jzz code and see if I can find the issue.
